sql/sql_expression.h

```diff
@@ -1332,6 +1308,10 @@ struct TableExpression: public std::enable_shared_from_this<TableExpression> {
     */
     virtual UnboundEntities getUnbound() const = 0;
 
+    /** Match join qualification (Inner, left, right, full)
```

Comment:
Do we need to expose this and JoinQualification as part of the API?  I would have them as free functions defined inside the class.